task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -108,3 +118,9 @@ def stop(self):
 
     def get_event_queue(self):
         return self.dest_queue
+
+    def grow(self, task_id):
+        return task_id.append(self.retries - self.task_retries[task_id])
+
+    def shrink(self, task_id):
+        return task_id[0:-1]
```

Comment:
enforcing a "no colon in id components" rule is fairly easy, tracking and mapping new ids for each retry is much harder and context is completely lost for every layer below retrying.